### PR TITLE
Introduce _get_image_ident and fix a bug that needed it

### PR DESCRIPTION
**Pull request by hkaj**

### CODE REVIEW

File: tests/core/test_service_discovery.py [hunk 1]
```diff
@@ -499,6 +499,7 @@ def test_get_auto_config(self):
             'consul': ('consul', None, {
                 "url": "http://%%host%%:%%port%%", "catalog_checks": True, "new_leader_checks": True
             }),
+            'redis:v1': ('redisdb', None, {"host": "%%host%%", "port": "%%port%%"}),
```
Comment (hkaj) [inline tests/core/test_service_discovery.py]: test that would have avoided the bug

File: utils/service_discovery/abstract_config_store.py [hunk 2]
```diff
@@ -62,9 +62,9 @@ def dump_directory(self, path, **kwargs):
         raise NotImplementedError()
 
     def _get_auto_config(self, image_name):
-        # use the image name, ignore the tag
-        if image_name.split(':')[0] in self.auto_conf_images:
-            check_name = self.auto_conf_images[image_name]
+        ident = self._get_image_ident(image_name)
+        if ident in self.auto_conf_images:
+            check_name = self.auto_conf_images[ident]
```
Comment (hkaj) [inline utils/service_discovery/abstract_config_store.py]: that was the bug leading to skipping a check that we could actually enable

File: utils/service_discovery/abstract_config_store.py [hunk 3]
```diff
@@ -125,17 +125,17 @@ def read_config_from_store(self, identifier):
             except KeyNotFound:
                 log.debug("Could not find directory {} in the config store, "
                           "trying to convert to the old format.".format(identifier))
-                legacy_ident = identifier.split(':')[0].split('/')[-1]
-                res = self._issue_read(legacy_ident)
+                image_ident = self._get_image_ident(identifier)
```
Comment (hkaj) [inline utils/service_discovery/abstract_config_store.py]: In this case it's used for backward compatibility so I didn't change the debug message. Same thing 10 lines below. We shouldn't rely on this as the main identifier as it's not fine grained enough.

**Comment by masci** [comment]
👍

**Comment by hkaj** [comment]
Merging, tests failures are unrelated. Thanks @masci !

**Pull request merged**